string-functions: Explain binary-as-hex behavior (#19117)
pingcap/docs: functions-and-operators/string-functions.md

- If the argument is a number, `HEX(n)` returns a hexadecimal string representation of `n`. The function treats the argument `n` as a `BIGINT` number, equivalent to using `CONV(n, 10, 16)`.
- If the argument is `NULL`, the function returns `NULL`.

> **Note:**
>

> > In the MySQL Client the [`--binary-as-hex`](https://dev.mysql.com/doc/refman/8.0/en/mysql-command-options.html#option_mysql_binary-as-hex) option is enabled by default in interactive mode, this causes the client to display data for which is doesn't know the character set as [`hexadecimal literal`](https://dev.mysql.com/doc/refman/8.0/en/hexadecimal-literals.html) instead. You can use the `--skip-binary-as-hex` option to disable this behavior.

Reviewer: ```suggestion
> In the MySQL Client the [`--binary-as-hex`](https://dev.mysql.com/doc/refman/8.0/en/mysql-command-options.html#option_mysql_binary-as-hex) option is enabled by default in interactive mode, this causes the client to display data with unknown character set as a [`hexadecimal literal`](https://dev.mysql.com/doc/refman/8.0/en/hexadecimal-literals.html). You can use the `--skip-binary-as-hex` option to disable this behavior.
```

> **Note:**
>
> - The argument must be a valid hexadecimal value that contains `0`–`9`, `A`–`F`, or `a`–`f`. If the argument is `NULL` or falls outside this range, the function returns `NULL`.

> > - In the MySQL Client the [`--binary-as-hex`](https://dev.mysql.com/doc/refman/8.0/en/mysql-command-options.html#option_mysql_binary-as-hex) option is enabled by default in interactive mode, this causes the client to display data for which is doesn't know the character set as [`hexadecimal literal`](https://dev.mysql.com/doc/refman/8.0/en/hexadecimal-literals.html) instead. You can use the `--skip-binary-as-hex` option to disable this behavior.

Reviewer: ```suggestion
> - In the MySQL Client the [`--binary-as-hex`](https://dev.mysql.com/doc/refman/8.0/en/mysql-command-options.html#option_mysql_binary-as-hex) option is enabled by default in interactive mode, this causes the client to display data with unknown character set as a [`hexadecimal literal`](https://dev.mysql.com/doc/refman/8.0/en/hexadecimal-literals.html). You can use the `--skip-binary-as-hex` option to disable this behavior.
```
